metamodel.zenta

```diff
@@ -2335,180 +2365,1111 @@ A role is assigned to human(s).
+* Don't use abbreviations.
+* The most specific the use of something, the longer its name.
+* Do not be redundant.
+
+</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="Focus on API" id="906e0159-1ab7-4ec0-b0a5-858519ab1f37" ancestor="behaviour">
+            <documentation>* Focus on API rather than patterns.
+* First, write down the API for perfect scenario, observe how it feels, then jump to coding and make it work.
+* If it helps understanding, extract expressions to local variables and/or methods.
+* Tell, don't ask.</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="No comments" id="55d0f60c-ceb7-456a-88ba-31fe434b6b4f" ancestor="behaviour">
+            <documentation>* If you need to write comments to explain your code, it means you need to put it in a new method.
+* Do not comment out code. Remove it.
+* The //NOPMD comment counts as annotatiion disabling code check</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="Disable checks only if you must" id="793c9f63-3124-4059-8223-35820b5371fc" ancestor="behaviour">
+            <documentation>* The annotations disabling code checks should be rare and explained
+* Exceptions of rules should always be thought out, explained and be available later.
+* The right place to explain is right in the annotation (or near it if there is no other way: this is an exception of the &quot;No comments&quot; rule).
+* If there are many exceptions, then either you do something wrong, or the rules should be changed: raise the issue so we can solve it
+                                  </documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="polymorphism" id="ff23e5db-5fae-4191-963c-83a556a74b5e" ancestor="behaviour">
+            <documentation>* If there is an if/else or switch/case, the designer might want to create specific classes for specific behaviour.
+* If there are flag argument to a function, there should be more functions instead.
+</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="Structure the code" id="68b5cde9-a64c-4e07-a0e2-2e0a3109acbb" ancestor="behaviour">
+            <documentation>Structure the code well. Separate concepts vertically.
+* Declare variables close to their usage.
+* Dependent functions should be close.
+* Similar functions should be close.
+* Place supporting functions below supported ones.
+* Use white space to associate related things and disassociate weakly related.
+                                     
+                                  </documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="KISS" id="a554b350-a076-44e8-96bb-083ca197c1fd" ancestor="behaviour">
+            <documentation>Keep it simple, stupid.
+* Extract the complex logic to other method.
+* Avoid the use of else.
```

Comment:
Overusing if-else can cause high Cyclomatic complexity, but it's not the root of all evil. Sometimes it helps to keep your code readable.